File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/singleton/static-init/singleton.ts
export class Singleton {
  private static count = 0;
  private static instance: Singleton;

  private static init = (_ => {
    console.log("Singleton Object STATIC BLOCK\n" +
        "Singleton Object Instance initialized");
    Singleton.instance = new Singleton();
  })();

  private constructor() {
    Singleton.count++;
  }

  public say(): void {
    console.log("Singleton: ", Singleton.count);
  }

  public static getInstance(): Singleton {
    return Singleton.instance;
  }
}

File: src/c#/Creational/Singleton/SingletonReflection/Singleton.cs
﻿using System;
namespace SingletonReflection
{
    public class Singleton
    {

        private static Singleton singleton = new Singleton();

        private static int count;
        private String name;

        private Singleton()
        {
            count++;
            name = "Singleton" + count;
        }

        public static Singleton Instance
        {
            get
            {
                return singleton;
            }
        }

        public void PrintName()
        {
            Console.WriteLine(name);
        }
    }
}
